atom/renderer/renderer_client_base.cc

```diff
@@ -81,24 +73,31 @@ std::vector<std::string> ParseSchemesCLISwitch(const char* switch_name) {
 }  // namespace
 
 RendererClientBase::RendererClientBase() {
+  auto* command_line = base::CommandLine::ForCurrentProcess();
   // Parse --standard-schemes=scheme1,scheme2
   std::vector<std::string> standard_schemes_list =
-      ParseSchemesCLISwitch(switches::kStandardSchemes);
+      ParseSchemesCLISwitch(command_line, switches::kStandardSchemes);
   for (const std::string& scheme : standard_schemes_list)
     url::AddStandardScheme(scheme.c_str(), url::SCHEME_WITHOUT_PORT);
   isolated_world_ = base::CommandLine::ForCurrentProcess()->HasSwitch(
       switches::kContextIsolation);
+  // We rely on the unique process host id which is notified to the
+  // renderer process via command line switch from the content layer,
+  // if this switch is removed from the content layer for some reason,
+  // we should define our own.
+  DCHECK(command_line->HasSwitch(::switches::kRendererClientId));
+  renderer_client_id_ =
+      command_line->GetSwitchValueASCII(::switches::kRendererClientId);
```

Comment:
can we use `RenderThread::GetClientId()` here rather than parsing it out manually?

Reply:
`RenderThread::Get` can only be used when the render thread has been started, it may not be available at this point also it may not be the same thread. To avoid confusion of which might get triggered first render thread creation or script context creation, I am just taking safety first approach here.

Reply:
`RenderThread` is created pretty early in initialization, even before the sandbox is enabled: https://chromium.googlesource.com/chromium/src/+/67.0.3396.127/content/renderer/renderer_main.cc#228

I don't see a way that `RendererClient` could be created before `RenderThread` is initialized.

Reply:
Er, wait, actually I think I was wrong. It definitely is called before `RenderThread` is initialized:

[content_main_runner.cc:543](https://chromium.googlesource.com/chromium/src/+/67.0.3396.127/content/app/content_main_runner.cc#543) is what calls `CreateContentRendererClient`, which will ultimately call this function.

[content_main_runner.cc:562](https://chromium.googlesource.com/chromium/src/+/67.0.3396.127/content/app/content_main_runner.cc#562) is what will initialize `RenderThread`.